src/test/java/com/iexec/common/sdk/order/payload/RequestOrderTest.java

```diff
@@ -0,0 +1,67 @@
+package com.iexec.common.sdk.order.payload;
+
+import com.iexec.common.chain.DealParams;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+
+class RequestOrderTest {
+
+    @Test
+    void shouldSerializeEmptyMapWhenNoParameters() {
+        DealParams params = DealParams.builder().build();
+        Assertions.assertEquals("{}", RequestOrder.toStringParams(params));
+    }
+
+    @Test
+    void shouldSerializeEmptyMapWhenBooleanParametersAreFalse() {
+        DealParams params = DealParams.builder()
+                .iexecResultEncryption(false)
+                .iexecDeveloperLoggerEnabled(false)
+                .build();
+        Assertions.assertEquals("{}", RequestOrder.toStringParams(params));
+    }
+
+    @Test
+    void shouldSerializeParametersWithEncryptedResult() {
+        for (String storageProxy : List.of(DealParams.DROPBOX_RESULT_STORAGE_PROVIDER, DealParams.IPFS_RESULT_STORAGE_PROVIDER)) {
+            String expectedData = "{\"iexec_result_encryption\":true";
+            expectedData += ",\"iexec_result_storage_proxy\":\"" + storageProxy + "\"}";
+            DealParams params = DealParams.builder()
+                    .iexecResultEncryption(true)
+                    .iexecResultStorageProxy(storageProxy)
+                    .build();
+            Assertions.assertEquals(expectedData, RequestOrder.toStringParams(params));
+        }
+    }
+
+    @Test
+    void shouldSerializeArgs() {
+        String expectedData = "{\"iexec_args\":\"arg1 arg2 arg3\"}";
+        DealParams params = DealParams.builder()
+                .iexecArgs("arg1 arg2 arg3")
+                .build();
+        Assertions.assertEquals(expectedData, RequestOrder.toStringParams(params));
+    }
+
+    @Test
+    void shouldSerializeInputFiles() {
+        String expectedData = "{\"iexec_input_files\":[\"file1\",\"file2\"]}";
+        DealParams params = DealParams.builder()
+                .iexecInputFiles(List.of("file1", "file2"))
+                .build();
+        Assertions.assertEquals(expectedData, RequestOrder.toStringParams(params));
+    }
+
+    @Test
+    void shouldSerializeSecrets() {
+        String expectedData = "{\"iexec_secrets\":{\"0\":\"0\"}}";
+        DealParams params = DealParams.builder()
+                .iexecSecrets(Map.of("0", "0"))
```

Comment:
I'd prefer to have two distinct values. Otherwise, there could be some switch between key and value and we could not notice that.

Reply:
Yes, plus a map having more than 1 mapping

Reply:
Fixed in https://github.com/iExecBlockchainComputing/iexec-common/pull/298/commits/7107a8db754fd4807bf1ec149afd78fa41506bd1